ktor-client/ktor-client-core/jsAndWasmShared/src/io/ktor/client/utils/HeadersUtils.kt

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2014-2024 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.utils
+
+import io.ktor.http.HeadersBuilder
+import io.ktor.http.HttpHeaders
+
+/**
+ * The body is already decompressed for wasm and js engine,
+ * so we're dropping these headers
+ * to make sure all the plugins and checks work with correct content length and encoding.
+ */
+internal fun HeadersBuilder.dropCompressionHeaders() {
+    if (contains(HttpHeaders.ContentEncoding)) {
+        remove(HttpHeaders.ContentEncoding)
+        remove(HttpHeaders.ContentLength)
+    }
+}
```

Comment:
This could interfere with any other introspection on the response headers... Maybe it would be a better solution to introduce some flag to the engine interface to indicate implicit decompression for the client to check.

Reply:
The problem is with content-length header has wrong value after browser decoding. Let's check it as a separate issue